src/x/server/server.go

```diff
@@ -155,11 +182,15 @@ func (s *server) serve() {
 		} else {
 			s.wgConns.Add(1)
 			go func() {
-				s.handler.Handle(conn)
-
-				conn.Close()
-				s.removeConnectionFn(conn)
-				s.wgConns.Done()
+				defer conn.Close()
+				defer s.removeConnectionFn(conn)
+				defer s.wgConns.Done()
+
+				securedConn, err := s.maybeUpgradeToTLS(conn)
+				if err != nil {
+					return
```

Comment:
Silent failure? What should be the handling if upgrade to TLS fails? Seems like it could be a warning, and then we proceed with non TLS? Or it could be fatal for the request, but either way, we need logs + metrics on this.

Reply:
Fixed